pyfakefs/fake_filesystem_shutil.py

```diff
@@ -285,7 +285,9 @@ def _destinsrc(src, dst):
             return dst
 
         source_is_dir = stat.S_ISDIR(self.filesystem.GetObject(src).st_mode)
-        if source_is_dir and self.filesystem.Exists(dst):
+        if source_is_dir:
+          dst = os.path.join(dst, os.path.basename(src))
+          if self.filesystem.Exists(dst):
```

Comment:
Please use `self.filesystem.JoinPaths` instead of `os.path.join` to consider a possibly changed path separator in the fake filesystem.